dvr-ws/routes/index.js

```diff
@@ -2,18 +2,35 @@ var express = require('express');
 var router = express.Router();
 var config = require('../../common/Configuration');
 var logger = require('../logger/logger');
-var qio = require('q-io/fs');
 var path = require('path')
 var persistenceFormat = require('../../common/PersistenceFormat');
+var fsUtils = require('../../lib/utils/fs-utils');
 
-router.get(/\/smil\:([^\\/]*)_all\.smil\/([^\?]*)/i, function(req, res) {
-
+router.get(/\/smil:([^\\/]*)_all\.smil\/([^\?]*)/i, function(req, res) {
     var entryId = req.params[0];
     var fileName = req.params[1];
 
-    var disk = path.join(persistenceFormat.getEntryFullPath(entryId),fileName);
-    res.sendFile(disk);
+    var fullPath = path.join(persistenceFormat.getEntryFullPath(entryId),fileName);
+
+    if ( fileName.search('chunklist.m3u8') > -1 ) {
+
+        // Todo: take the configuration initialization outside
+        var chunklistExpireAge = config.get("webServerParams:chunklistExpireAge");
+
+        fsUtils.checkIsFileExpired(fullPath, chunklistExpireAge)
+            .then(function () {
+                res.sendFile(fullPath);
+            })
+            .catch(function () {
+                var message = fileName + '%s not found';
+                res.status(404).send(message);
```

Comment:
do we need content?  look what wowza response and akamai as well to figure out the best